src/celeritas/optical/Interaction.hh

```diff
@@ -0,0 +1,92 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/Interaction.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "geocel/Types.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * The result of a discrete optical interaction.
+ *
+ * Discrete optical interactions are always elastic. The wavelength of a
+ * a photon is only changed through absorption re-emission processes.
+ */
+struct Interaction
+{
+    //! Interaction result category
+    enum class Action
+    {
+        scattered,  //!< Still alive, state has changed
+        absorbed,  //!< Absorbed by the material
+        unchanged,  //!< No state change, no secondaries
+        failed,  //!< Ran out of memory during sampling
+    };
+
+    Real3 direction;  //!< Post-interaction direction
+    Real3 polarization;  //!< Post-interaction polarization
```

Comment:
Correct, the first iteration let's go for absolute minimum functionality (as long as it doesn't prevent us from adding the new capabilities going forward).

Reply:
We don't have to decide in this PR, but if the track initialization stuff will be similar to what's in our main stepping loop it's possible it will be easier to include secondaries from the start rather than removing that logic and adding it back in later.

Reply:
Perhaps... looking at the code though, the "locate tracks" executor uses core params and physics step view, and the initialization-from-secondaries is likewise complicated...